### ISSUE TITLE
Icon preview

### ISSUE THREAD

**Issue by sonichy** [openscad/openscad#4602]
After install FlashPrint, stl files have icon preview in windows explorer.
Can OpenSCAD have icon preview in scad files ?
![alt](https://www.hualigs.cn/image/643a39eade442.jpg)

**Comment by tpimh** [comment]
Yes, there's a script for thumbnailing .scad files: [OpenSCAD Thumbnailer](https://srlm.io/2015/12/15/scad-thumbnailer/).
Also one for .stl files: [OpenSCAD STL Thumbnailer](https://www.thingiverse.com/thing:258653). It can also be adapted to support other formats that OpenSCAD can import.

**Comment by jordanbrown0** [comment]
Remember that it can take an arbitrary amount of time to preview an OpenSCAD file.  Usually it's just a few seconds, but it can be many seconds or longer.  A malicious program could run forever.

It isn't at all clear that it's a good idea to spend that time creating thumbnails by default.

And of course there are plenty of .SCAD files that are intended only for `include`, not for executing directly.

**Comment by AJolly** [comment]
> Yes, there's a script for thumbnailing .scad files: [OpenSCAD Thumbnailer](https://srlm.io/2015/12/15/scad-thumbnailer/). Also one for .stl files: [OpenSCAD STL Thumbnailer](https://www.thingiverse.com/thing:258653). It can also be adapted to support other formats that OpenSCAD can import.
Note that thats for linux, not windows like the user originally asked.  I have yet to be able to find a way to generate thumbnails for openscad files on windows.  

And for time taken, thats what the OS's own thumbnail cache is for (and if you want, pregenerate the thumbnails)
